### ISSUE TITLE
Scalar construction in C++: `torch.tensor(7)` !~= `at::tensor(7)`

### ISSUE THREAD

**Issue by crutcher** [pytorch/pytorch#63897]
## 🐛 Bug

The documented way to construct a scalar in python `torch` is `torch.tensor(X)` for scalar values of X; which produces a Rank-0 (sizes == []) Tensor with one 0-dim value `X`.

a10 doesn't really ... have documentation; it seems to largely leverage familiarity with the python semantics.

The equivalent interface in a10 is `at::tensor(X)` for scalar values of X; which produces ... a Rank-1 (sizes = [1]) Tensor with one 1-dim value `{X}`. 

This seems ... wrong?
